libs/community/langchain_community/document_loaders/directory.py

```diff
@@ -197,3 +198,55 @@ def load(self) -> List[Document]:
             pbar.close()
 
         return docs
+
+    def lazy_load(self) -> Iterator[Document]:
+        """Load documents lazily."""
+        p = Path(self.path)
+        if not p.exists():
+            raise FileNotFoundError(f"Directory not found: '{self.path}'")
+        if not p.is_dir():
+            raise ValueError(f"Expected directory, got file: '{self.path}'")
+
+        paths = p.rglob(self.glob) if self.recursive else p.glob(self.glob)
+        items = [
+            path
+            for path in paths
+            if not (self.exclude and any(path.match(glob) for glob in self.exclude))
+        ]
+
+        if self.sample_size > 0:
+            if self.randomize_sample:
+                randomizer = random.Random(
+                    self.sample_seed if self.sample_seed else None
+                )
+                randomizer.shuffle(items)
+            items = items[: min(len(items), self.sample_size)]
+
+        for i in items:
+            yield from self.lazy_load_file(i, p)
+
+    def lazy_load_file(self, item: Path, path: Path) -> Iterator[Document]:
```

Comment:
Could you make this private? This is a non standard part of the document loader API so generally we do not expect users to be relying on it.

Could you also update load_file to proxy to this method, so we can avoid code duplication?

Reply:
Renamed lazy_load_file to _lazy_load_file. Made load proxy to lazy_load. load_file wasn't used anymore, so I deleted it.